### PR TITLE
chore: Update tools-to-drawer treatment

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Fixing merge logic between separate tools slot and drawers.

Behavior in the current public API

```js
<AppLayout /> // renders empty tools drawer [1]
<AppLayout tools="content" /> // renders tools drawer with content [2]
<AppLayout toolsHide={true} /> // hides tools drawer
```

How this will change when `drawers` public property is introduced

```js
<AppLayout drawers={[]} /> // renders one drawer for the tools (looks identical to [1] from above)
<AppLayout tools="content" drawers={[]} /> // same as [2] from above
<AppLayout tools="some tools" drawers={[drawer]} /> // renders two drawers, the first one is "tools"
```

Related links, issue #, if available: n/a

### How has this been tested?

Updated existing unit tests and added a few extra ones for corner-cases

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__tests__/drawers.test.tsx [hunk 1]
```diff
@@ -33,11 +33,18 @@ describeEachAppLayout(() => {
...
-  test('renderds drawers with the tools', () => {
-    const { wrapper } = renderComponent(<AppLayout tools="Test" {...singleDrawer} />);
+  test('should apply drawers treatment to the tools if at least one other drawer is provided', () => {
+    const { wrapper } = renderComponent(<AppLayout contentType="form" {...singleDrawer} />);
+    expect(wrapper.findDrawersTriggers()).toHaveLength(2);
+    expect(wrapper.findToolsToggle()).toBeFalsy();
+  });
+
+  test('renders drawers with the tools', () => {
+    const { wrapper } = renderComponent(<AppLayout contentType="form" {...singleDrawer} />);
```
Comment (katiegeorge) [inline src/app-layout/__tests__/drawers.test.tsx:47]: Do we need this test now with the one above?

Comment (just-boris) [reply src/app-layout/__tests__/drawers.test.tsx:47]: True, they test same use-case.

There will be merge conflicts with https://github.com/cloudscape-design/components/pull/1527. I will review these tests one more time when resolving the merge conflict

**Review by taheramr — APPROVED** [review]
looks good to me
